File: src/CorporateMessengerWithAgile/Application/Dto/EmployeeWithCompanyAndPositionDto.cs
namespace Application.Dto
{
    public class EmployeeWithCompanyAndPositionDto : BaseDto
    {
        public CompanyDto Company { get; set; } = null!;
        public PositionInCompanyDto PositionInCompany { get; set; } = null!;
        public UserDto User { get; set; } = null!;
        public IReadOnlyList<TeamMemberDto> TeamMembers { get; set; } = [];
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/EmployeeWithCompanyAndPositionDto.ts
// src/models/entity/EmployeeWithCompanyAndPositionDto.ts
import { BaseDto } from "./BaseDto";
import { Guid } from "../Guid";
import { CompanyDto } from "./CompanyDto";
import { PositionInCompanyDto } from "./PositionInCompanyDto";
import { UserDto } from "./UserDto";
import { TeamMemberDto } from "./TeamMemberDto";

export class EmployeeWithCompanyAndPositionDto extends BaseDto {
    public company: CompanyDto;
    public positionInCompany: PositionInCompanyDto;
    public user: UserDto;
    public teamMembers: TeamMemberDto[];

    constructor(
        id: Guid,
        company: CompanyDto,
        positionInCompany: PositionInCompanyDto,
        user: UserDto,
        teamMembers: TeamMemberDto[]
    ) {
        super(id);
        this.company = company;
        this.positionInCompany = positionInCompany;
        this.user = user;
        this.teamMembers = teamMembers;
    }
}
